packages/executor/src/index.ts

```diff
@@ -37,89 +44,140 @@ export class ChugSplashExecutor extends BaseServiceV2<Options, Metrics, State> {
         ChugSplashManagerABI,
         signer
       )
 
+      // get active bundle id for this project
       const activeBundleId = await manager.activeBundleId()
       if (activeBundleId === ethers.constants.HashZero) {
-        console.log('no active bundle')
+        this.logger.error(`Error: No active bundle id found in manager`)
         continue
       }
 
+      // fetch bundle state
       const bundleState: ChugSplashBundleState = await manager.bundles(
         activeBundleId
       )
 
-      // TODO: Add this to the ChugSplashManager contract
-      const selectedExecutor = await manager.getSelectedExecutor(activeBundleId)
-      if (selectedExecutor !== ethers.constants.AddressZero) {
-        // Someone else has been selected to execute the upgrade, so we can skip it.
-        continue
-      }
-
+      // get proposal event and compile
       const proposalEvents = await manager.queryFilter(
         manager.filters.ChugSplashBundleProposed(activeBundleId)
       )
-
-      if (proposalEvents.length !== 1) {
-        // TODO: throw an error here or skip
-      }
-
       const proposalEvent = proposalEvents[0]
       const { bundle, canonicalConfig } = await compileRemoteBundle(
         hre,
         proposalEvent.args.configUri
       )
+
+      // ensure compiled bundle matches proposed bundle
       if (bundle.root !== proposalEvent.args.bundleRoot) {
-        // TODO: throw an error here or skip
+        // log error and continue
+        this.logger.error(
+          'Error: Compiled bundle root does not match proposal event bundle root',
+          canonicalConfig.options
+        )
+        continue
+      }
+
+      // execute bundle
+      try {
+        await hre.run('chugsplash-execute', {
+          chugSplashManager: manager,
+          bundleState,
+          bundle,
+          parsedConfig: canonicalConfig,
+          deployer: signer,
+          hide: false,
+        })
+      } catch (e) {
+        // log error and continue
+        this.logger.error('Error: execution error', e, canonicalConfig.options)
+        continue
       }
 
-      // todo call chugsplash-execute if deploying locally
-      await hre.run('chugsplash-execute', {
-        chugSplashManager: manager,
-        bundleState,
-        bundle,
-        parsedConfig: canonicalConfig,
-        deployer: signer,
-        hide: false,
-      })
-
-      if ((await getChainId(this.state.wallet.provider)) !== 31337) {
-        await verifyChugSplashConfig(hre, proposalEvent.args.configUri)
+      // verify on etherscan
+      try {
+        if ((await getChainId(this.state.wallet.provider)) !== 31337) {
+          await verifyChugSplashConfig(hre, proposalEvent.args.configUri)
+        }
+      } catch (e) {
+        this.logger.error(
+          'Error: verification error',
+          e,
+          canonicalConfig.options
+        )
+      }
+
+      this.logger.info('Successfully executed and verified')
```

Comment:
It'd be nice to separate this into two separate logs: a 'Successfully executed' log that's called after `chugsplash-execute` above, and a 'Successfully verified' log that's called here. This way it's easier to distinguish errors caused during execution and errors caused during verification. I expect the latter to fail pretty often since Etherscan's API can be flakey